tasks/broad/ImputationBeagleTasks.wdl

```diff
@@ -0,0 +1,217 @@
+version 1.0
+
+task CountVariantsInChunks {
+  input {
+    File vcf
+    File vcf_index
+    File panel_bed_file
+
+    String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.5.0.0"
+    Int cpu = 1
+    Int memory_mb = 16000
+    Int disk_size_gb = 2 * ceil(size([vcf, vcf_index, panel_bed_file], "GiB")) + 10
+  }
+  Int command_mem = memory_mb - 1500
+  Int max_heap = memory_mb - 1000
+
+  command <<<
+    set -e -o pipefail
+
+    ln -sf ~{vcf} input.vcf.gz
+    ln -sf ~{vcf_index} input.vcf.gz.tbi
+
+    gatk --java-options "-Xms~{command_mem}m -Xmx~{max_heap}m" CountVariants -V input.vcf.gz | tail -n 1 > var_in_original
+    bedtools intersect -a ~{vcf} -b ~{panel_bed_file} | wc -l > var_also_in_reference
+  >>>
+
+  output {
+    Int var_in_original = read_int("var_in_original")
+    Int var_also_in_reference = read_int("var_also_in_reference")
+  }
+  runtime {
+    docker: gatk_docker
+    disks: "local-disk ${disk_size_gb} HDD"
+    memory: "${memory_mb} MiB"
+    cpu: cpu
+    preemptible: 3
+  }
+}
+
+task CheckChunks {
+  input {
+    Int var_in_original
+    Int var_also_in_reference
+
+    String bcftools_docker = "us.gcr.io/broad-gotc-prod/imputation-bcf-vcf:1.0.7-1.10.2-0.1.16-1669908889"
+    Int cpu = 1
+    Int memory_mb = 4000
+  }
+  command <<<
+    set -e -o pipefail
+
+    if [ $(( ~{var_also_in_reference} * 2 - ~{var_in_original})) -gt 0 ] && [ ~{var_also_in_reference} -gt 3 ]; then
+      echo true > valid_file.txt
+    else
+      echo false > valid_file.txt
+    fi
+  >>>
+  output {
+    Boolean valid = read_boolean("valid_file.txt")
+  }
+  runtime {
+    docker: bcftools_docker
+    disks: "local-disk 10 HDD"
+    memory: "${memory_mb} MiB"
+    cpu: cpu
+    preemptible: 3
+  }
+}
+
+task Phase {
+  input {
+    File dataset_vcf
+    File ref_panel_bref3
+    File genetic_map_file
+    String basename
+    String chrom             # not needed if ref file has been chunked and you are using the entire chunk
```

Comment:
if these are not needed, should they be optional inputs?

Reply:
ah these were comments from our collaborator when we were creating this WDL; they are indeed needed. I'll remove these comments since they're confusing.